src/table/components/head/__tests__/ColumnAdjuster.spec.tsx

```diff
@@ -29,32 +35,104 @@ describe('<ColumnAdjuster />', () => {
         qApprMaxGlyphCount: 10,
         columnWidth: { type: ColumnWidthTypes.AUTO },
       } as Column,
+      {
+        label: 'col2',
+        pageColIdx: 1,
+        qApprMaxGlyphCount: 10,
+        columnWidth: { type: ColumnWidthTypes.AUTO },
+      } as Column,
     ];
     column = columns[0];
     isLastColumn = false;
-    rootElement = {
-      getBoundingClientRect: () => ({ height: 100 } as DOMRect),
-    } as HTMLElement;
     applyColumnWidths = jest.fn();
+    columnWidths = [200, 200];
+    setColumnWidths = jest.fn();
     constraints = {};
   });
 
-  it('should return null when applyColumnWidths is undefined', () => {
-    applyColumnWidths = undefined;
+  afterEach(() => jest.clearAllMocks());
+
+  it('should return null when  constraints.active is true', () => {
+    constraints.active = true;
 
     renderAdjuster();
     expect(screen.queryByTestId('sn-table-column-adjuster')).toBeNull();
   });
 
-  it('should return null when constraints.active is true', () => {
-    constraints.active = true;
+  it('should render when constraints.active is not true', () => {
+    renderAdjuster();
+    expect(screen.queryByTestId('sn-table-column-adjuster')).toBeInTheDocument();
+  });
 
+  it('should change column width using keyboard', async () => {
     renderAdjuster();
-    expect(screen.queryByTestId('sn-table-column-adjuster')).toBeNull();
+    const columnAdjuster = screen.queryByTestId('sn-table-column-adjuster') as HTMLElement;
+
+    fireEvent.keyDown(columnAdjuster, { key: KeyCodes.RIGHT });
+    await waitFor(() => {
+      expect(setColumnWidths).toHaveBeenNthCalledWith(1, [205, 200]);
+    });
+
+    fireEvent.keyDown(columnAdjuster, { key: KeyCodes.SPACE });
+    await waitFor(() => {
+      expect(applyColumnWidths).toHaveBeenNthCalledWith(1, { type: ColumnWidthTypes.PIXELS, pixels: 205 }, column);
+    });
   });
 
-  it('should render adjuster component', () => {
+  it('should not change column width when confirming with the same column width and not when canceling ', async () => {
     renderAdjuster();
-    expect(screen.queryByTestId('sn-table-column-adjuster')).toBeInTheDocument();
+    const columnAdjuster = screen.queryByTestId('sn-table-column-adjuster') as HTMLElement;
+
+    fireEvent.keyDown(columnAdjuster, { key: KeyCodes.SPACE });
+    await waitFor(() => {
+      expect(applyColumnWidths).toHaveBeenCalledTimes(0);
+    });
+
+    fireEvent.keyDown(columnAdjuster, { key: KeyCodes.ESC });
+    await waitFor(() => {
+      expect(applyColumnWidths).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  it('should change column width using mouse', async () => {
```

Comment:
I think this probably a good case for integration test not unit tests. IMO unit tests should not be responsible for visual testing.

same for below test too

Reply:
Not sure I agree in this case. This is a pretty hard file to test. A component test on this level seems to work fine. dragging in integration tests feels very flaky to me. Potentially we could do this in an integration test later, but I think this is at least a good compromise for now